subsys/bluetooth/adv_prov/providers/fast_pair.c

```diff
@@ -41,37 +57,97 @@ int bt_le_adv_prov_fast_pair_set_battery_mode(enum bt_fast_pair_adv_battery_mode
 	return 0;
 }
 
-static int get_data(struct bt_data *ad, const struct bt_le_adv_prov_adv_state *state,
-		    struct bt_le_adv_prov_feedback *fb)
+static enum bt_fast_pair_adv_mode get_adv_mode(bool pairing_mode)
 {
+	enum bt_fast_pair_adv_mode adv_mode;
+
+	if (pairing_mode) {
+		adv_mode = BT_FAST_PAIR_ADV_MODE_DISCOVERABLE;
+	} else {
+		if (show_ui_pairing) {
+			adv_mode = BT_FAST_PAIR_ADV_MODE_NOT_DISCOVERABLE_SHOW_UI_IND;
+		} else {
+			adv_mode = BT_FAST_PAIR_ADV_MODE_NOT_DISCOVERABLE_HIDE_UI_IND;
+		}
+	}
+
+	return adv_mode;
+}
+
+static int generate_payload(enum bt_fast_pair_adv_mode adv_mode)
```

Comment:
Consider passing the `adv_battery_mode` (to avoid using global provider state implicitly here)